tests/js/test_compile_shell.py

```diff
@@ -0,0 +1,173 @@
+        # PARTICULAR PURPOSE.  THE CODE PROVIDED HEREUNDER IS ON AN "AS IS" BASIS,
+        # AND THERE IS NO OBLIGATION WHATSOEVER TO PROVIDE MAINTENANCE,
+        # SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+
+        def assertLogs(self, logger=None, level=None):  # pylint: disable=missing-return-doc,missing-return-type-doc
+
+            _LoggingWatcher = collections.namedtuple("_LoggingWatcher", ["records", "output"])
+
+            class _CapturingHandler(logging.Handler):
+                def __init__(self):
+                    logging.Handler.__init__(self)
+                    self.watcher = _LoggingWatcher([], [])
+
+                def emit(self, record):
+                    self.watcher.records.append(record)
+                    self.watcher.output.append(self.format(record))
+
+            class _AssertLogsContext(object):  # pylint: disable=too-few-public-methods
+                LOGGING_FORMAT = "%(levelname)s:%(name)s:%(message)s"
+
+                def __init__(self, test_case, logger_name, level):
+                    self.test_case = test_case
+                    self.logger = None
+                    self.logger_name = logger_name
+                    self.level = getattr(logging, level) if level else logging.INFO
+                    self.msg = None
+                    self.old = None
+                    self.watcher = None
+
+                def __enter__(self):  # pylint: disable=missing-return-doc,missing-return-type-doc
+                    if isinstance(self.logger_name, logging.Logger):
+                        self.logger = self.logger_name
+                    else:
+                        self.logger = logging.getLogger(self.logger_name)
+                    handler = _CapturingHandler()
+                    handler.setFormatter(logging.Formatter(self.LOGGING_FORMAT))
+                    self.watcher = handler.watcher
+                    self.old = (self.logger.handlers[:], self.logger.propagate, self.logger.level)
+                    self.logger.handlers = [handler]
+                    self.logger.setLevel(self.level)
+                    self.logger.propagate = False
+                    return handler.watcher
+
+                def __exit__(self, exc_type, exc_value, tb):  # pylint: disable=missing-return-doc
+                    # pylint: disable=missing-return-type-doc
+                    self.logger.handlers, self.logger.propagate = self.old[:2]
+                    self.logger.setLevel(self.old[2])
+                    if exc_type is not None:
+                        return False
+                    self.test_case.assertGreater(
+                        len(self.watcher.records), 0,
+                        "no logs of level %s or higher triggered on %s" % (
+                            logging.getLevelName(self.level), self.logger.name))
+
+            return _AssertLogsContext(self, logger, level)
+
+
+class CompileTests(TestCase):
```

Comment:
Looks like a good start. BTW you can also look at the [more recent FuzzManager tests](https://github.com/MozillaSecurity/FuzzManager/blob/master/Collector/test_Collector.py#L268) for examples of pytest format tests which I find are simpler.